core/network/src/protocol.rs

```diff
@@ -728,7 +728,7 @@ impl<B: BlockT, S: NetworkSpecialization<B>, H: ExHashT> Protocol<B, S, H> {
 			match request.direction {
 				message::Direction::Ascending => id = BlockId::Number(number + One::one()),
 				message::Direction::Descending => {
-					if number.is_zero() {
+					if number.is_one() {
```

Comment:
if it is zero, will that cause an error further down the line when `parent_hash` gives `None` for the header/block?

Reply:
`parent_hash` is not an `Option`. for the genesis block it is a zero hash.

Reply:
That change is still wrong though. We should allow downloading the genesis block in principle.